Lesson2/Lesson2/BankAccount.cs

```diff
@@ -88,8 +88,32 @@ public void WithdrawFromBalance(double sum)
         public double Balance { get; set; }
         public TypeAccount TypeAccount { get; set; }
 
-        public void PrintAccountInfo()
+        public override bool Equals(object obj)
         {
+            return obj is BankAccount account &&
+                   ClientAccountNumber == account.ClientAccountNumber &&
+                   Balance == account.Balance &&
+                   TypeAccount == account.TypeAccount;
+        }
+
+        public override int GetHashCode()
+        {
+            return HashCode.Combine(ClientAccountNumber, Balance, TypeAccount);
+        }
+
+        static public bool operator ==(BankAccount a, BankAccount b)
+        {
+            return a.ClientAccountNumber == b.ClientAccountNumber;
```

Comment:
Здесь должен быть вызов метода Equals
```suggestion
            return Equals(a, b);
```